bot/exts/valentines/lovecalculator.py

```diff
@@ -22,45 +24,41 @@
 class LoveCalculator(Cog):
     """A cog for calculating the love between two people."""
 
+    @in_month(Month.FEBRUARY)
     @commands.command(aliases=("love_calculator", "love_calc"))
     @commands.cooldown(rate=1, per=5, type=commands.BucketType.user)
-    async def love(self, ctx: commands.Context, who: Union[Member, str], whom: Union[Member, str] = None) -> None:
+    async def love(self, ctx: commands.Context, who: Member, whom: Optional[Member] = None) -> None:
         """
         Tells you how much the two love each other.
 
-        This command accepts users or arbitrary strings as arguments.
-        Users are converted from:
+        This command requires at least one member as input, if two are given love will be calculated between
+        those two users, if only one is given, the second member is asusmed to be the invoker.
+        Members are converted from:
           - User ID
           - Mention
           - name#discrim
           - name
           - nickname
 
-        Any two arguments will always yield the same result, though the order of arguments matters:
-          Running .love joseph erlang will always yield the same result.
-          Running .love erlang joseph won't yield the same result as .love joseph erlang
-
-        If you want to use multiple words for one argument, you must include quotes.
-          .love "Zes Vappa" "morning coffee"
+        Any two arguments will always yield the same result, regardless of the order of arguments:
+          Running .love @joe#6000 @chrisjl#2655 will always yield the same result.
+          Running .love @chrisjl#2655 @joe#6000 will yield the same result as before.
         """
         if whom is None:
             whom = ctx.author
 
-        def normalize(arg: Union[Member, str]) -> Coroutine:
-            if isinstance(arg, Member):
-                # If we are given a member, return name#discrim without any extra changes
-                arg = str(arg)
-            else:
-                # Otherwise normalise case and remove any leading/trailing whitespace
-                arg = arg.strip().title()
-            # This has to be done manually to be applied to usernames
-            return clean_content(escape_markdown=True).convert(ctx, arg)
+        if not all((
+            Lovefest.role_id in [role.id for role in who.roles],
+            Lovefest.role_id in [role.id for role in whom.roles]
+        )):
+            raise BadArgument("Both members must have the lovefest role!")
```

Comment:
Might be worth mentioning how to get the lovefest role, something like
```suggestion
            raise BadArgument("Both members must have the lovefest role! You can get this by running the `.lovefest sub` command.")
```
if the `who` or `whom` is the invoker we probably wouldn't need to require the lovefest role, but the current behaviour is fine.